### ISSUE TITLE
Max of only 500 pages are returned

### ISSUE THREAD

**Issue by amitjoy** [atlassian-api/atlassian-python-api#1380]
while executing - `atlassian.confluence.Confluence.get_all_pages_from_space` with a `limit` set to `2000` still returns only 500 pages whereas the space has more than 500 pages. Is it a restriction? How can I retrieve all the pages setting the limit properly?

**Comment by Spacetown** [comment]
That's the limit of the server if you want to get more pages you need to define the start.

@gonchik I think it's really time to implement the paging in a generic way. The only problem is that the function than will return an generator instead of a list which can break custom scripts.

**Comment by gkowalc** [comment]
@Spacetown I agree, but this will be a major change that can break many things, so maybe this is something we can add to next major (4.x) version. For now, as a workaround maybe we should edit the docs to highlight the need of implementing pagination  with few examples?

**Comment by Spacetown** [comment]
I tried yesterday to do the pagination in the rest client and configuring it in the module for the existing once.

**Comment by gonchik** [comment]
There I have provided method with generator 
https://github.com/atlassian-api/atlassian-python-api/commit/0e7b9bfca1772c0bb2850b7723fc36ba4ab2feb6
